data_structures/monotonic_stack.cpp

```diff
@@ -0,0 +1,51 @@
+#include <iostream>
+#include <vector>
+
+class MonotonicStack {
+private:
+    std::vector<int> stack;
+
+public:
+    // Push an element while maintaining the monotonic property
+    void push(int value) {
```

Comment:
You can `inline` all your methods. (ref: https://google.github.io/styleguide/cppguide.html#Inline_Functions)